docs/src/references/hsm-support.md

```diff
@@ -1,242 +1,227 @@
+- store the corresponding certificate on the file system
+- set `device.cert_path` to the certificate path
+- set `device.cryptoki.module_path` to the correct PKCS #11 dynamic library
+- set `device.cryptoki.pin` and `device.cryptoki.uri` accordingly to the local HSM settings
 
-8. Configure the opensc-pkcs11 module for p11-kit server and client
+:::
 
+### Part 1: Setup the cryptographic token
+
+1. Install SoftHSM2 (to create the token and key) and `p11tool` (to view the [PKCS #11 URI][p11uri]
+   of the key).
+
+    ```sh
+    sudo apt-get install -y softhsm2 gnutls-bin
+    ```
+
+    [p11uri]: https://www.rfc-editor.org/rfc/rfc7512
+
+For SoftHSM configuration, see [SoftHSM README](https://github.com/softhsm/softHSMv2?tab=readme-ov-file#configure-1).
+
+2. Add tedge and current user to `softhsm` group. Only users belonging to `softhsm` group can view
+   and manage SoftHSM tokens. After adding your own user, remember to logout and login for changes
+   to take effect. Alternatively, you can just run `softhsm2-util` and `p11tool` with `sudo`.
+
+   ```sh
+    sudo usermod -a -G softhsm tedge
+    sudo usermod -a -G softhsm $(id -un)
+   ```
+
+3. Create a new SoftHSM token. You'll be prompted for a PIN for a regular user and security officer
+   (SO). The rest of the guide assumes PIN=123456, but you're free to use a different one.
+
+    ```sh
+    softhsm2-util --init-token --slot 0 --label my-token
+    ```
+
+4. Import the private key to the created token. Make sure to use the correct PIN value for a regular
+   user from the previous step.
+
+    ```sh
+    PUB_PRIV_KEY=$(
+        cat "$(tedge config get device.key_path)" && cat "$(tedge config get device.cert_path)"
+    )
+    softhsm2-util \
+        --import <(echo "$PUB_PRIV_KEY") \
+        --token my-token \
+        --label my-key \
+        --id 01 \
+        --pin 123456 \
+    ```
+
+5. Get the URI of the key
+
+    First, see what tokens are available
+
+    ```sh
+    p11tool --list-tokens
+    ```
+
+    ```sh title="p11tool --list-tokens"
+    ...
+    Token 2:
+        URL: pkcs11:model=SoftHSM%20v2;manufacturer=SoftHSM%20project;serial=83f9cf49039c051a;token=my-token
+        Label: my-token
+        Type: Generic token
+        Flags: RNG, Requires login
+        Manufacturer: SoftHSM project
+        Model: SoftHSM v2
+        Serial: 83f9cf49039c051a
+        Module: /usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so
+    ...
+    ```
+
+    Now check if the private key object is in the token. You may need to login, provide the regular
+    user PIN and also provide token URL(URI) if multiple tokens are connected:
+
+    ```sh
+    sudo p11tool --login --set-pin=123456 --list-privkeys "pkcs11:model=SoftHSM%20v2;manufacturer=SoftHSM%20project;serial=83f9cf49039c051a;token=my-token"
+    ```
+    ```sh title="sudo p11tool --login --set-pin=123456 --list-privkeys "pkcs11:model=SoftHSM%20v2;manufacturer=SoftHSM%20project;serial=83f9cf49039c051a;token=my-token""
```

Comment:
I'm pretty sure the title was broken anyway due to the nested double quotes...but easier just to use "Output" as the title.

````suggestion
    ```sh title="Output"
````

Reply:
fixed in 3eebdb3945